recirq/quantum_chess/circuit_transformer.py

```diff
@@ -323,8 +319,7 @@ def qubit_mapping(self,
             mapping[cur_node] = start_qubit
             start_list.remove(start_qubit)
 
-        if len(mapping) != len(g):
-            print('Warning: could not map all qubits!')
+        assert len(mapping) == len(g) + len(sq)
```

Comment:
Should we assert with a message so that people get a message?
Alternatively, should this raise an appropriate Exception instead?

Reply:
There is always supposed to be a one-to-one mapping between named qubits and allocated grid qubits. An assertion seems appropriate to me since if it does not hold it means there is a bug in qubit_mapping.

Added a message to the assertion